subprojects/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/TaskSubclassingBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -213,4 +215,45 @@ apply plugin: SomePlugin
         version previous requireGradleDistribution() withTasks 'assemble' inDirectory(file("producer")) run()
         version current requireGradleDistribution() withTasks 't' run()
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/6027")
+    def "task can use project.file() from statically typed Groovy"() {
+        // Don't run these for RC 3, as stuff did change during the RCs
+        Assume.assumeFalse(previous.version == GradleVersion.version("2.14-rc-3"))
+
+        when:
+        file("producer/build.gradle") << """
+            apply plugin: 'groovy'
+            dependencies {
+                ${previous.version < GradleVersion.version("1.4-rc-1") ? "groovy" : "compile"} localGroovy()
```

Comment:
Do we really want to test versions this old? Shouldn't we annotate this class with something like `@TargetVersions(["3.0+"])`?

Reply:
There is no such thing as `@TargetVersions("3.0+")` unfortunately. And the tests run just fine on 1.0, so why not keep them? In any case, if we want to cull the number of cross-version tests we run, that should happen on a general basis, not on a single test.

Reply:
I thought there was: https://github.com/gradle/gradle/blob/6bd97fbe62de4180a8ad946f9d61edaac9eb09d7/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/CrossVersionTestRunner.groovy#L70-L73

I'm fine with running this test for every version.

Reply:
Hm. When I added that annotation it just failed with some exception. In any case, I still think we should not reduce the range of tested versions for some tests but not others.